### PR TITLE
LLP GenFilter based on transverse displacement and pdgId

### PR DESCRIPTION
**Pull request by Raffaella07**
#### PR description:

The filter generalizes MCLongLivedParticles GenFilter (filter was included in GenFilters up toCMSSW_11_0_X/ , 
see  https://github.com/cms-sw/cmssw/blob/CMSSW_11_0_X/GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc 
It selects particles based on transverse displacement (by means of a lower and/or upper cut) and pdgIds.
The filter is developed for searches of long lived HNLs in B meson decays (@rmanzoni  @pandolf  @amlyon)

The filter is included in the `plugins` subdirectory of `GeneratorInterface/GenFilters` . 
It is possible to retrieve the original filter behavior by setting the lower cut only and leaving the pdgIds unspecified.
NOTE: the filter syntax has changed with respect to the original version, as a lower (LengMin) and an upper (LengMax) cut can be specified, while in the CMSSW_11_0_X/  version there was only a lower cut parameter.

#### PR validation:

Filter behavior can be tested in the `test` subdirectory of  `GeneratorInterface/GenFilters` through the module `test_MCLongLivedFilter_cfg.py`

Backported to CMSSW_10_2_X with #37684

### CODE REVIEW

**Comment by SiewYan** [comment]
please test

**Comment by amlyon** [comment]
Dear experts, 

Now that all the checks have passed, is there anything else to be done on our side? 

If not, could you please proceed to the merge of this PR? 

Thanks a lot!

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        if ((dist2 >= theLowerCut2 || theLowerCut <= 0.) &&
+            (dist2 < theUpperCut2 ||
+             theUpperCut <= 0.)) {  //lower cut can be also 0 - prompt particle needs to be accepted in that case
+          pass = true;
+          break;
+        }
+      }
+
+      if (((*p)->production_vertex() == nullptr) && (!((*p)->end_vertex() != nullptr))) {
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:71]:
```suggestion
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      matchedID = false;
+      for (unsigned int idx = 0; idx < particleIDs.size(); idx++) {
+        if (abs((*p)->pdg_id()) == abs(particleIDs.at(idx))) {  //compares absolute values of pdgIds
+          matchedID = true;
+          break;
+        }
+      }
+    }
+
+    if (matchedID) {
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:56]:
```suggestion
    if (matchedID) {
       if {theLowerCut <= 0. && theUpperCut <= 0.)  {
          pass = true;
          break;
       }
```

**Review by perrotta — COMMENTED** [review]
Please also define a `fillDescriptions(...)` method for your filter

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Pythia8/Pythia.h"
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h:35]: What is this included header needed for?

File: GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py [hunk 4]
```diff
@@ -53,8 +53,11 @@
 )
 
 process.select = cms.EDFilter("MCLongLivedParticles",
```
[the remark this answers is not in the document]
Comment (Raffaella07) [reply GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py:55]: I believe that this tester module is a residue of the CMSSW versions in which the filter existed. In fact, I couldn't find a MCLongLivedParticles filter in CMSSW_12_4_X, but it existed up until CMSSW_11_0_X/ ,
see https://github.com/cms-sw/cmssw/blob/CMSSW_11_0_X/GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc 
The CMSSW_10_2_X backport for this filter in fact is being merged to a pre-existing version of the filter: https://github.com/cms-sw/cmssw/pull/37684/files

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+using namespace edm;
+using namespace std;
+
+//Filter particles based on their minimum and/or maximum displacement on the transverse plane and optionally on their pdgIds
+//To run independently of pdgId, do not insert the particleIDs entry in filter declaration
+
+MCLongLivedParticles::MCLongLivedParticles(const edm::ParameterSet& iConfig)
+    : token_(consumes<edm::HepMCProduct>(
+          edm::InputTag(iConfig.getUntrackedParameter("moduleLabel", std::string("generator")), "unsmeared"))),
```
[the remark this answers is not in the document]
Comment (Raffaella07) [reply GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:15]: I noticed that most the other filters in place used the token based initialization, but I can revert it back of course

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  float theUpperCut2 = theUpperCut * theUpperCut;
+  float theLowerCut2 = theLowerCut * theLowerCut;
+
+  const HepMC::GenEvent* generated_event = evt->GetEvent();
+  HepMC::GenEvent::particle_const_iterator p;
+
+  for (p = generated_event->particles_begin(); p != generated_event->particles_end(); p++) {
+    //if a list of pdgId is provided, loop only on particles with those pdgId
+    if (particleIDs.at(0) != 0) {
```
[the remark this answers is not in the document]
Comment (Raffaella07) [reply GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:46]: It is actually to explicitly check whether the particleIDs vector has been initialized by the user with a different value with respect to the default (the default setting of the filter reproduces the behavior of the old MCLongLivedParticleFilter  https://github.com/cms-sw/cmssw/blob/CMSSW_11_0_X/GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc where no particleID was selectable ). 
I'll move the check out of the loop, great suggestion, thanks!

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+  bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+private:
+  // ----------member data ---------------------------
+  std::string moduleLabel_;
+  edm::InputTag hepMCProductTag_;
+  const edm::EDGetTokenT<edm::HepMCProduct> token_;
+  std::vector<int>
+      particleIDs;  //possible now to chose on which pdgIds the filter is applied - if ParticleIDs.size()==0 runs on all particles in  the event as the preovious filter version
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h:51]:
```suggestion
 // To chose on which pdgIds the filter is applied - if ParticleIDs.size()==0 runs on all particles in  the event
  std::vector<int> particleIDs_; 
```
i.e.
- Put the comment in a separate line so that the code-checks do not split it, and remove reference to a previous filter which is not currently present in cmssw;
- Use a trailing underscore as for all other class members

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace edm {
+  class HepMCProduct;
+  class ConfigurationDescriptions;
+}  // namespace edm
+
+class MCLongLivedParticles : public edm::global::EDFilter<> {
+public:
+  explicit MCLongLivedParticles(const edm::ParameterSet&);
+  // ~MCLongLivedParticles() override;
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h:39]:
```suggestion
  ~MCLongLivedParticles() override = default;
```
(why did you remove it)

File: GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py [hunk 9]
```diff
@@ -53,8 +53,11 @@
...
-    hepMCProductTag = cms.InputTag("VtxSmeared"),
-    LengCut = cms.untracked.double(100.0) ## in mm
+   # MinLeng = cms.untracked.double(100.0) ## in mm
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py:56]: Please remove unneeded (and non-existing) parameters)

File: GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py [hunk 10]
```diff
@@ -53,8 +53,11 @@
...
+    #hepMCProductTag = cms.untracked.string(""),
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/test/test_MCLongLivedFilter_cfg.py:57]: This is needed, otherwise the default "moduleLabel" is kept

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+      hepMCProductTag_(
+          edm::InputTag(iConfig.getUntrackedParameter(moduleLabel_, std::string("generator")), "unsmeared")),
+      token_(consumes<edm::HepMCProduct>(hepMCProductTag_)),
+      particleIDs(iConfig.getParameter<std::vector<int>>("ParticleIDs")),
+      theUpperCut_(iConfig.getParameter<double>("LengMax")),
+      theLowerCut_(iConfig.getParameter<double>("LengMin")) {}
+
+void MCLongLivedParticles::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("hepMCProductTag", "moduleLabel");
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:27]:
```suggestion
  desc.add<std::string>("hepMCProductTag", "moduleTag");
```
if you want to use a generic name as default, perhaps something which includes "Tag" instead of "Lablìel" could be preferred
Even better if, instead of"moduleTag" you write the actual tag, e.g. (taking the same example as in your L19 above):
```C++
  desc.add<std::string>("hepMCProductTag", ("moduleLabel","generator","unsmeared));
```
(where in place of "moduleLabel" you should write the actual product that you want to get)

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+#include <vector>
+
+using namespace edm;
+using namespace std;
+
+//Filter particles based on their minimum and/or maximum displacement on the transverse plane and optionally on their pdgIds
+//To run independently of pdgId, do not insert the particleIDs entry in filter declaration
+
+MCLongLivedParticles::MCLongLivedParticles(const edm::ParameterSet& iConfig)
+    : moduleLabel_(iConfig.getParameter<std::string>("hepMCProductTag")),
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:17]: Since you use hepMCProductTag_ this label is not needed any more

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+      hepMCProductTag_(
+          edm::InputTag(iConfig.getUntrackedParameter(moduleLabel_, std::string("generator")), "unsmeared")),
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:19]:
```suggestion
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+private:
+  // ----------member data ---------------------------
+  std::string moduleLabel_;
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h:47]:
```suggestion
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+                          (((*p)->production_vertex())->position().y() - ((*p)->end_vertex())->position().y());
+
+        if ((dist2 >= theLowerCut2 || theLowerCut_ <= 0.) &&
+            (dist2 < theUpperCut2 ||
+             theUpperCut_ <= 0.)) {  //lower cut can be also 0 - prompt particle needs to be accepted in that case
+          pass = true;
+          break;
+        }
+      }
+      if (((*p)->production_vertex() == nullptr) && (!((*p)->end_vertex() != nullptr))) {
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:82]:
```suggestion
```
This can remain in the same `if block` as above
(But perhaps you wanted to check only the production vertex above, and the end vertex here... Please check)

**Review by perrotta — COMMENTED** [review]
Could you also please also the header and the implementation file in the same .cc, and remove the header file afterwards?

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc (hunk 15 above)
Comment (Raffaella07) [reply GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:82]: Yes, sorry, I tested the code with and without the check, it is needed to prevent from running on particles missing the prod vertex

**Comment by SiewYan** [comment]
please test

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:131]:
```suggestion
      if (((*p)->production_vertex() == nullptr) && ((*p)->end_vertex() == nullptr)) {
```
This is what you coded with the two negations.
Maybe you wanted to accept events with `(*p)->end_vertex() != nullptr` instead

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+      }
+      if (((*p)->production_vertex() != nullptr) && ((*p)->end_vertex() != nullptr)) {
+        float dist2 = (((*p)->production_vertex())->position().x() - ((*p)->end_vertex())->position().x()) *
+                          (((*p)->production_vertex())->position().x() - ((*p)->end_vertex())->position().x()) +
+                      (((*p)->production_vertex())->position().y() - ((*p)->end_vertex())->position().y()) *
+                          (((*p)->production_vertex())->position().y() - ((*p)->end_vertex())->position().y());
+
+        if ((dist2 >= theLowerCut2 || theLowerCut_ <= 0.) &&
+            (dist2 < theUpperCut2 ||
+             theUpperCut_ <= 0.)) {  //lower cut can be also 0 - prompt particle needs to be accepted in that case
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:126]:
```suggestion
        // lower/upper cuts can be also <= 0 - prompt particle needs to be accepted in that case
        if ((theLowerCut_ <= 0. || dist2 >= theLowerCut2) &&
            (theUpperCut_ <= 0. || dist2 < theUpperCut2)) {  
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+
+  Handle<HepMCProduct> evt;
+
+  iEvent.getByToken(token_, evt);
+
+  bool pass = false;
+  bool matchedID = true;
+
+  float theUpperCut2 = theUpperCut_ * theUpperCut_;
+  float theLowerCut2 = theLowerCut_ * theLowerCut_;
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:95]:
```suggestion
  const float theUpperCut2 = theUpperCut_ * theUpperCut_;
  const float theLowerCut2 = theLowerCut_ * theLowerCut_;
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+            (dist2 < theUpperCut2 ||
+             theUpperCut_ <= 0.)) {  //lower cut can be also 0 - prompt particle needs to be accepted in that case
+          pass = true;
+          break;
+        }
+      }
+      if (((*p)->production_vertex() == nullptr) && (!((*p)->end_vertex() != nullptr))) {
+        if ((((*p)->end_vertex()->position().perp() >= theLowerCut_) || theLowerCut_ <= 0.) &&
+            (((*p)->end_vertex()->position().perp() < theUpperCut_) ||
+             theUpperCut_ <= 0.)) {  // lower cut can be also 0 - prompt particle needs to be accepted in that case
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:134]:
```suggestion
        // lower/upper cuts can be also 0 - prompt particle needs to be accepted in that case
        float distEndVert = (*p)->end_vertex()->position().perp();
        if ((theLowerCut_ <= 0. || distEndVert >= theLowerCut_) &&
            (theUpperCut_ <= 0. || distEndVert < theUpperCut_)) {  
```

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+
+  if (particleIDs_.at(0) != 0)
+    matchedID = false;
+
+  const HepMC::GenEvent* generated_event = evt->GetEvent();
+  HepMC::GenEvent::particle_const_iterator p;
+
+  for (p = generated_event->particles_begin(); p != generated_event->particles_end(); p++) {
+    //if a list of pdgId is provided, loop only on particles with those pdgId
+
```
Comment (perrotta) [inline GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:105]: Remove this empty line (as the comment above refers to what's here below

File: GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+    : hepMCProductTag_(iConfig.getParameter<edm::InputTag>("hepMCProductTag")),
+      token_(consumes<edm::HepMCProduct>(hepMCProductTag_)),
+      particleIDs_(iConfig.getParameter<std::vector<int>>("ParticleIDs")),
+      theUpperCut_(iConfig.getParameter<double>("LengMax")),
+      theLowerCut_(iConfig.getParameter<double>("LengMin")) {}
+
+//Filter description
+void MCLongLivedParticles::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("hepMCProductTag", edm::InputTag("generator", "unsmeared"));
```
[the remark this answers is not in the document]
Comment (Raffaella07) [reply GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:76]: I want to add the input tag with its default to the filter description, is this the correct way to do it? would you do it in another way?

[the remark this answers is not in the document]
Comment (perrotta) [reply GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc:76]: You use that tag to get a HepMCProduct from the Event: which is the exact name of the product that you want to get? This is what should end in your "hepMCProductTag"

**Comment by Raffaella07** [comment]
Hey everyone, can we please retest the latest commit? We would like to try to get the filter in for the friday production release

**Comment by perrotta** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
@Raffaella07 please provide some evidence that the latest version of the code actually runs and filters according to the original intention

**Comment by Raffaella07** [comment]
Here are the output of a running example ([outFull.txt](https://github.com/cms-sw/cmssw/files/8632187/outFull.txt)) and  some plots to show the filter behavior.
[FilterValid_plots.pdf](https://github.com/cms-sw/cmssw/files/8632181/FilterValid_plots.pdf)

**Comment by SiewYan** [comment]
please test

**Comment by amlyon** [comment]
Dear experts, 

Once the tests are completed and if there are no comments left, could you please proceed to the merge in the best delays? 

I would then point you to the backport of this PR to 10_2 [#37684](https://github.com/cms-sw/cmssw/pull/37684).

Thanks a lot!

**Comment by perrotta** [comment]
Dear @amlyon and @Raffaella07

This PR still needs to be scrutinized by @cms-sw/generators-l2, so we must in any case wait for their review before proceeding. Once fully accepted by all stakeholders I see no issues for merging it in master.

About a possible backport to 10_2, let me remind you that all plans for the releases are discussed and decided at the ORP meeting every Tuesday, next one being planned for May 10 [1]. I would have given for granted that your proposal had been already discussed within the GEN group, and then brought to PPD and PDMV, which is where all those productions must be coordinated. Unfortunately even if requested explicitely about it at the last ORP meeting it does not seem that GEN was fully aware of the request (which is probably the reason why this PR was not yet reviewed by GEN in the 11 days since it was submitted). Let me suggest you to make sure that either any of you or someone that is aware of your request can be present at the release planning meeting when it has to be discussed.

So, please agree your plans first together with GEN and PPD; then the release managers will follow in putting the release related part of those plans in practice. 

[1]  https://indico.cern.ch/event/1156305/

**Comment by amlyon** [comment]
Dear @perrotta, 

Thank you very much for this clarification, this is precisely the information that we requested to be able conduct the effort in the smoothest way.

Let me share with you what are our intentions now:

- Given the differences between the master and backported versions of the code, we would like to follow your suggestion of renaming the filter, such that we would avoid backward compatibility issues with the already existing version in 10_2_X. This means
   - Keep the code as is in the master (modulo renaming of the class)
   - Propagate the exact same code to 10_2_X (modulo the release-specific items that have to be adapted such that the code can run + the fact that both the .h and .cc files are needed)
- Get in touch with the GEN experts and PPD to ask for their approval

Would you agree with such a plan? Do you think that this would be enough to converge, or will there be further steps to be undertaken before the release can be created? 

Thank you again for your collaboration.

**Comment by perrotta** [comment]
@amlyon that's fine with me, modulo that if you backport a different class (different name and file) I don't think you need to customize it specifically for 12_2; in particular, you **don't** need to split the single file in the \plugin area into tow .h and .cc files in separate folders

**Comment by Raffaella07** [comment]
> @amlyon that's fine with me, modulo that if you backport a different class (different name and file)

-Yes we will backport the filter with the same **new** name we will use for the master.

> I don't think you need to customize it specifically for 12_2; in particular, you **don't** need to split the single file in the \plugin area into two .h and .cc files in separate folders

In CMSSW_10_2_X there is no /plugin area, there's an /interface directory for .h files and an /src directory for .cc files, all the filters are implemented such that the header is in /interface and the implementation in /src. Should we move everything into one file in /src nonetheless?  (we can discuss this afterwards, but still, to have a clear picture on our side  ) And more importantly, can we do this in the PRs we have already open or should we close them and open different ones?

**Comment by perrotta** [comment]
@Raffaella07 just create a new \plugin area, and add your filter there (only your filter, of course, no need to bother about the other ones). 

Yes, you could do everything already in this PR... But I think it will be easier and less troublesome to close this PR, wait that the one in the master converges and it is ready to get merged, and then make a new PR by simply copying what you finally submitted in the master.

**Comment by amlyon** [comment]
Thanks @perrotta for your guidance. We will proceed with your suggestions.

File: GeneratorInterface/GenFilters/test/test_MCDisplacementFilter_cfg.py [hunk 22]
```diff
@@ -52,9 +52,10 @@
     )
 )
 
-process.select = cms.EDFilter("MCLongLivedParticles",
-    hepMCProductTag = cms.InputTag("VtxSmeared"),
-    LengCut = cms.untracked.double(100.0) ## in mm
+process.select = cms.EDFilter("MCDisplacementFilter",
+    ParticleIDs = cms.vint32(310), ## in mm
```
Comment (Saptaparna) [inline GeneratorInterface/GenFilters/test/test_MCDisplacementFilter_cfg.py:56]: Please fix the comment here. It can be confusing. 310 is the PDGID of the $K_s^{0}$.

Comment (amlyon) [reply GeneratorInterface/GenFilters/test/test_MCDisplacementFilter_cfg.py:56]: @perrotta, do you think that you could make a commit to remove the legend "#in mm" on line 56? I have tried to do it myself, but do not have the rights to push a commit, and @Raffaella07 is currently not available. Thank you very much in advance!

**Comment by amlyon** [comment]
> 

Thank you very much @Saptaparna for the prompt review!

File: GeneratorInterface/GenFilters/test/test_MCDisplacementFilter_cfg.py (hunk 22 above)
Comment (perrotta) [reply GeneratorInterface/GenFilters/test/test_MCDisplacementFilter_cfg.py]: > @perrotta, do you think that you could make a commit to remove the legend "#in mm" on line 56? I have tried to do it myself, but do not have the rights to push a commit, and @Raffaella07 is currently not available. Thank you very much in advance!

Done: apparently I have the superpowers to do it :-)

**Comment by amlyon** [comment]
Many thanks @perrotta for the quick fix! Does the PR look alright to you now? If so, we'll wait for the tests to be done in this PR and will then open the new backport PR with the exact same code, as discussed earlier.

**Comment by perrotta** [comment]
please test

**Comment by Saptaparna** [comment]
+1
